pages/api/ws.js

```diff
@@ -374,8 +375,30 @@ const ioHandler = (req, res) => {
             let game = rooms[message.room].game;
             let copy_round = game.round;
             let copy_title = game.title;
+
+            // Save only the hidden states from incoming data
+            const incomingHiddenStates = {};
+            if (message.data.registeredPlayers) {
+              Object.entries(message.data.registeredPlayers).forEach(([id, player]) => {
+                // Skip if player is "host" or not an object
+                if (typeof player === 'object' && player !== null && typeof player.hidden === 'boolean') {
+                  incomingHiddenStates[id] = { hidden: player.hidden };
+                }
+              });
+            }
+
+            // Remove registered players data from client message to prevent unauthorized player modifications
             delete message.data.registeredPlayers;
+
```

Comment:
Needs to be implemented in the new backend.

About here is where the player struct object is so you can add a new field.
https://github.com/joshzcold/Cold-Family-Feud/blob/4ddc2088d2ceec30c11ce453fe7808a32a8ea883/backend/api/game.go#L13

Here is where you can add new actions
https://github.com/joshzcold/Cold-Family-Feud/blob/4ddc2088d2ceec30c11ce453fe7808a32a8ea883/backend/api/pipe.go#L28

Here is where a player is created.
NOTE: in golang structs the default value if not specified for boolean would be `false` automatically.
https://github.com/joshzcold/Cold-Family-Feud/blob/4ddc2088d2ceec30c11ce453fe7808a32a8ea883/backend/api/session.go#L186

Reply:
The function to handle the generic `data` action is here
https://github.com/joshzcold/Cold-Family-Feud/blob/4ddc2088d2ceec30c11ce453fe7808a32a8ea883/backend/api/data.go#L24

Reply:
Added Hidden field now works same as with old backend